firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -349,13 +349,23 @@ private String readExistingIidOrCreateFid(PersistedInstallationEntry prefs) {
   /** Registers the created Fid with FIS servers and update the persisted state. */
   private PersistedInstallationEntry registerFidWithServer(PersistedInstallationEntry prefs)
       throws IOException {
+
+    // Note: Default value of instanceIdMigrationAuth: null
+    String iidToken = null;
+
+    if (prefs.getFirebaseInstallationId().equals(iidStore.readIid())) {
```

Comment:
Isn't reading from iidStore a complicated action?
Maybe checking the length of the identifier (e.g. == 11) is enough?
But in the end I don't care, just sharing my thoughts.